include/ff.h

```diff
@@ -173,7 +173,11 @@ typedef struct {
 	LBA_t	bitbase;		/* Allocation bitmap base sector */
 #endif
 	LBA_t	winsect;		/* Current sector appearing in the win[] */
-	BYTE	win[FF_MAX_SS];	/* Disk access window for Directory, FAT (and file data at tiny cfg) */
+#ifdef FS_FATFS_WIN_ALIGNED

```

Comment:
Wouldn't it work to always default `FS_FATFS_WIN_ALIGNED` to 1 when not set otherwise and remove the ifdef completely?

Reply:
Hi Ermel,

I was using the FS_FATFS_WIN_ALIGNED as the alignment value as well. See line 177.

In Zephyr, there is a config setting to define CONFIG_FS_FATFS_WIN_ALIGNED=512, meaning to align the win variable with 512 bytes, if setting the CONFIG_FS_FATFS_WIN_ALIGNED=16, it will align the win variable with 16 bytes.

Regards, 
Boon Khai

Reply:
what I mean is that if you do something like:
```
#if defined(CONFIG_FS_FATFS_WIN_ALIGNED)
#define FS_FATFS_WIN_ALIGNED CONFIG_FS_FATFS_WIN_ALIGNED
#else
#define FS_FATFS_WIN_ALIGNED 1
#fi
```
then is the `#ifdef still needed`?
or can it be just `BYTE win[FF_MAX_SS] __attribute__((aligned(FS_FATFS_WIN_ALIGNED)))`
where default value of 1 for `FS_FATFS_WIN_ALIGNED` would mean "no alignment".

Also, this seems problematic because if you set the alignment to 512 you may end up with a lot of wasted space in the structure.

Reply:
Hi Ermel, what I did is, (refer to ffconf.h)

#if defined(CONFIG_FS_FATFS_WIN_ALIGNED)
#if CONFIG_FS_FATFS_WIN_ALIGNED != 0
#define FS_FATFS_WIN_ALIGNED   CONFIG_FS_FATFS_WIN_ALIGNED
#endif /* CONFIG_FS_FATFS_WIN_ALIGNED != 0 */
#endif 

If the setting CONFIG_FS_FATFS_WIN_ALIGNED was not set, it will be 0 by default. When it is zero, then FS_FATFS_WIN_ALIGNED will not defined. When FS_FATFS_WIN_ALIGNED is not defined, the original definition of BYTE win[FF_MAX_SS] will be used without alignment. 

From our side, the SDMMC driver is copied block by block and need to be 512 byte aligned, and that's why I implement this CONFIG for user to choose whether or not to align the WIN. For those who don't have the hard requirement, they will not enable this CONFIG thus saving the space in the structure.

Reply:
Can you check how does the `__attribute__((aligned(1)));` behaves, because if it makes default alignment of byte array then the `#ifdef FS_FATFS_WIN_ALIGNED` s not needed: it would be enough, in such case, to do `BYTE win[FF_MAX_SS] __attribute__((aligned(FS_FATFS_WIN_ALIGNED)))` with no ifdef, because default value of `FS_FATFS_WIN_ALGNED=1` would work fine.

Reply:
> Can you check how does the `__attribute__((aligned(1)));` behaves, because if it makes default alignment of byte array then the `#ifdef FS_FATFS_WIN_ALIGNED` s not needed: it would be enough, in such case, to do `BYTE win[FF_MAX_SS] __attribute__((aligned(FS_FATFS_WIN_ALIGNED)))` with no ifdef, because default value of `FS_FATFS_WIN_ALGNED=1` would work fine.

Hi Ermel,

My bad, `__attribute__((aligned(1)));` actually works.
I agree with you suggestion to set default the value with 1, and with the following changes in the ffconf.h
```
#if defined(CONFIG_FS_FATFS_WIN_ALIGNED)
#define FS_FATFS_WIN_ALIGNED CONFIG_FS_FATFS_WIN_ALIGNED
#else
#define FS_FATFS_WIN_ALIGNED 1
#endif
```